src/panels/lovelace/components/hui-image.ts

```diff
@@ -206,7 +206,9 @@ export class HuiImage extends LitElement {
               : undefined,
         })}
         class="container ${classMap({
-          ratio: useRatio || this._lastImageHeight === undefined,
+          ratio:
+            useRatio ||
+            (this._lastImageHeight === undefined && imageSrc === undefined),
```

Comment:
Why would this change fix the issue? I think this change will break the intention of this code, as it will now only work for camera stream?

Reply:
The original code (before 2022.11) was just `ratio: useRatio` and there was no flickering, I tried to ensure that the new part (`this._lastImageHeight`) was not triggered for static images.

This, at least on my system, removes the flickering (that was very visible when multiple static images are stacked one over the other) that was also mentioned by other users but I admittedly don't know if there are negative side-effects.

Reply:
What I want to know is, what causes the flickering?

As when the image is loaded the first time, ratio should not do anything anymore?

Reply:
I think the flickering is related to some browser optimization (or lack of), when using the developer mode it does flicker a lot with any code I tried (I'm always talking about loaded images) and so I haven't been able to track down what really happens.

I actually don't understand why `ratio` should be true if `this._lastImageHeight` is undefined, specifically in case of static images. Maybe there is some different fix (ie. a proper one) to ensure that all the cases are handled properly (ie. during image load, after image load, camera stream, static images, etc.).

Reply:
ratio is true during loading, so the layout jumps at little as possible (ofc that is not possible to prevent completely) so when the image had not loaded yet, and we don't know the height we say it should display as 4:3. As soon as the image is loaded, we get the height of the image and use that instead of the ratio, also when the image later fails to load, we will keep the card the height of the previous image.

Reply:
btw, I can not replicate this issue, so if you can share your config, I can take a look at the issue.

Reply:
I tried with this simple config:
```
type: picture-elements
image: >-
  http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
elements:
type: picture-elements
image: >-
  http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
elements:
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 1%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 2%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 3%
      top: 0%
      left: 0%
      width: 200%
  - type: image
    image: >-
      http://hs.sbcounty.gov/CN/Photo%20Gallery/_t/Sample%20Picture%20-%20Koala_jpg.jpg?Mobile=0
    style:
      opacity: 1000000%
      top: 0%
      left: 0%
      width: 200%
```
I edited it from the UI and kept changing the opacity of the last image (by adding or removing more 0s, eg 1000%, 10000%, etc.), this is obviously not my actual use-case but I guess the effect is the same, you can clearly see it flickering (the multiple images are just required to make this more evident and frequent, depends on your browser/computer performance I guess).

This doesn't happen so often with `ratio: useRatio` and disappears completely if you also remove this piece of code a few lines above (which I also recently tried to fix in https://github.com/home-assistant/frontend/pull/14813):
```
            : this._lastImageHeight === undefined
            ? "56.25%"
```

Also, from Chrome developer's tools (again, during UI editing, adding/removing 0s in opacity just to trigger a refresh):
<img width="1348" alt="Screenshot 2022-12-28 at 15 23 58" src="https://user-images.githubusercontent.com/63957397/209828124-caf3008c-6088-4649-aaef-dd125b02b78e.png">
Image is loaded but ratio is true and, as you said, it shouldn't.

Reply:
Yes, if the config is changed it will recreate and cause a flicker, in dev mode that will be more visible as the cache is disabled and it takes longer for the image too load, but that should not happen in normal use cases?

Reply:
I discovered the issue using `custom:config-template-card` which might, in fact, recreate the image, hence the flickering (but, again, it didn't happen before 2022.11). 

In the end the whole "flickering issue" seems to be related to the fact that a static image is rendered with `ratio: true` during loading but I believe it shouldn't (since there is no ratio specified and since a static image does not have a proper loading phase).

Following what you said and having had a deeper look at `hui-image.ts`, I think that it contains too much logic specific to camera images mixed to generic images logic, which creates unexpected interactions (all the issues with `picture-elements` started with https://github.com/home-assistant/frontend/pull/14245 that should not affect anything after loading and should only affect video streams).

For example, this bit of code:

```
    if (this._loadState === LoadState.Loading && !this.cameraImage) {
      this._loadState = LoadState.Loaded;
    }
```

can be changed to:

```
    if (this._loadState === LoadState.Loading && !this.cameraImage) {
      this._loadState = LoadState.Loaded;
      this._lastImageHeight = 0;
    }
```

and this also seems to fix the flickering issue.

If this is a more acceptable fix for you, should I push this change?

Reply:
I think the usage in picture elements is the problem, so all the changes you make should only apply to that use case.

Reply:
Frankly I don't understand your point...

The issue (as shown above with the screenshot) is that static images are rendered with `ratio: true` while they have no actual ratio. As far as I can see, this is more evident in heavy usage of `custom:config-template-card` and also while editing a simple `picture-elements`, but the issue is there in any case.

My one-line-fix (or "workaround", in fact) doesn't seem to have negative side effects unlike https://github.com/home-assistant/frontend/pull/14245 for example (whose improvement on camera images broke static images). 

If any other fix won't be accepted, don't you think https://github.com/home-assistant/frontend/pull/14245 should be rolled back since its intent ("Prevents jumping of the UI during loading of video streams") affects different use cases (basically all the usages of static images) as demonstrated above?

Reply:
BTW: this interesting discussion and analysis clearly shows that `config-template-card` is not doing a good job since it keeps recreating its inner element, so I am migrating all my config to `card-mod` which doesn't seem affected by this issue (and, I guess, is faster and less demanding on the browser).

Reply:
> so I am migrating all my config to `card-mod` which doesn't seem affected by this issue (and, I guess, is faster and less demanding on the browser).

Card-mod seems to be working on a server side, CTC is working on a browser side. 
CTC may be replaced by card-mod for styling purpose; for some cases it is still cannot be replaced…